### ISSUE TITLE
any updates?

### ISSUE THREAD

**Issue by bubnenkoff** [monokles/telegram-d#1]
subj

**Comment by monokles** [comment]
Hi!

Sorry for not getting this done before the new year (as was my original plan)!
I'm still planning on continuing this in the very near future, but currently I'm moving a lot between countries, and do not have the time to sit down and work on it. 
I expect to be able to pick up the project again around the end of January.

The order in which I want to extend telegram-d  is as follows:

1.  Support file uploads
2.  Support webhook functionality
3. Extend API to current version

**Comment by bubnenkoff** [comment]
Thanks for fast answer! Could you add example of simple chat-bot?

**Comment by monokles** [comment]
Yup, I can include some examples once I continue working on it

**Comment by bubnenkoff** [comment]
Hi! Is there any news?


сб, 14 янв. 2017 г. в 11:58, Koen <notifications@github.com>:

**Comment by monokles** [comment]
Yeah, I took a look at the source code last week and was planning on implementing file upload support.
However, it happens that vibe-d does not support HTTP client requests with multipart/form-data yet, so I either need to implement that in vibe-d or figure out a workaround. Until then I cannot implement file upload support.

I will look at the other two issues in the coming weeks

**Comment by bubnenkoff** [comment]
The main problem is examples. For now I need only basic functionality, but without examples is very hard :(
you can look at https://github.com/ikod/dlang-requests afaik it have support multipart/form-data

**Comment by monokles** [comment]
Ok, thanks for the link to dlang-requests, I might look into using that instead. However, I think I will focus on getting web hooks support working first.

I'll try to write at least some simple examples this weekend.

**Comment by monokles** [comment]
Alright, I added two basic examples, you can run them by executing
` ./basic_bot.d <your API token>`
` ./callback_query.d <your API token>`

I tried to explain stuff a bit in the source but I kinda made them in a quick way so if there's something you're confused about just let me know.

I'm currently looking into generating documentation but ddoc isn't playing nice

**Comment by bubnenkoff** [comment]
Big thanks! I will look at it in next day!
